src/swarm-api/SwarmExtension.ts

```diff
@@ -0,0 +1,83 @@
+import { sendExtensionMessage } from '../messaging/e2e.messaging';
+
+async function sendSwarmExtensionMessage<Response>(
+  extensionId: string,
+  action: string,
+  sessionId?: string,
+  parameters?: unknown
+): Promise<Response> {
+  const { data, error } = await sendExtensionMessage<{
+    data: Response;
+    error: string;
+  }>(extensionId, { action, sessionId, parameters });
+
+  if (error) {
+    throw new Error(error);
+  }
+
+  return data;
+}
```

Comment:
Actually these two functions have different usage. The `sendExtensionMessage` function is a generic function that can be used with any extension, whereas the `sendSwarmExtensionMessage` should be used only with the Swarm extension. Because it extracts data and error objects and convert them into Promise.